src/python/foglamp/device/http_south_device.py

```diff
@@ -0,0 +1,188 @@
+# -*- coding: utf-8 -*-
+
+# FOGLAMP_BEGIN
+# See: http://foglamp.readthedocs.io/
+# FOGLAMP_END
+
+"""HTTP Listener handler for sensor readings"""
+import sys
+from aiohttp import web
+import asyncio
+from foglamp import logger
+from foglamp.web import middleware
+from foglamp.device.ingest import Ingest
+
+__author__ = "Amarendra K Sinha"
+__copyright__ = "Copyright (c) 2017 OSIsoft, LLC"
+__license__ = "Apache 2.0"
+__version__ = "${VERSION}"
+
+_LOGGER = logger.setup(__name__)
+
+_CONFIG_CATEGORY_NAME = 'http_south'
+_CONFIG_CATEGORY_DESCRIPTION = 'South Plugin HTTP Listener'
+_DEFAULT_CONFIG = {
+    'plugin': {
+         'description': 'http_south_device',
+         'type': 'string',
+         'default': 'http_south'
+    },
+    'port': {
+        'description': 'Port to listen on',
+        'type': 'integer',
+        'default': '6683',
+    },
+    'host': {
+        'description': 'URI to accept data on',
+        'type': 'string',
+        'default': '0.0.0.0',
+    },
+    'uri': {
+        'description': 'URI to accept data on',
+        'type': 'string',
+        'default': 'sensor-reading',
+    }
+}
+
+
+def plugin_info():
+    return {'name': 'http_south', 'version': '1.0', 'mode': 'async', 'type': 'device',
+            'interface': '1.0', 'config': _DEFAULT_CONFIG}
+
+
+def plugin_init(config):
+    """Registers HTTP Listener handler to accept sensor readings"""
+
+    _LOGGER.info("Retrieve HTTP Listener Configuration %s", config)
+
+    host = config['host']['value']
+    port = config['port']['value']  # Make port dynamic? its unattended env! expose as service port?
+    uri = config['uri']['value']
+
+    return {'host': host, 'port': port, 'uri': uri}
+
+
+def plugin_start(data):
+    try:
+        host = data['host']
+        port = data['port']
+        uri = data['uri']
+
+        loop = asyncio.get_event_loop()
+
+        app = web.Application(middlewares=[middleware.error_middleware])
+        app.router.add_route('POST', '/{}'.format(uri), HttpSouthIngest.render_post)
+        handler = app.make_handler()
+        coro = loop.create_server(handler, host, port)
+        server = asyncio.ensure_future(coro)
+
+        data['app'] = app
+        data['handler'] = handler
+        data['server'] = server
+    except Exception as e:
+        _LOGGER.exception(str(e))
+        sys.exit(1)
+
+
+def plugin_reconfigure(config):
+    pass
+
+
+def plugin_shutdown(data):
+    try:
+        app = data['app']
+        handler = data['handler']
+        server = data['server']
+
+        server.close()
+        asyncio.ensure_future(server.wait_closed())
+        asyncio.ensure_future(app.shutdown())
+        asyncio.ensure_future(handler.shutdown(60.0))
```

Comment:
not sure, it should be hard-coded or configurable  ?

Reply:
This calls for an architecture level decision and discussion and this issue can be taken up separately.